### PR TITLE
DDIC: Improve active/inactive check

### PR DESCRIPTION
**Pull request by mbtools**
Use `is_active_ddic` (function `DDIF_STATE_GET`) for all DDIC object types (in addition to `TABL`). 

This is more accurate than the current method (function `RS_INACTIVE_OBJECTS_WARNING`) since it detects newly created but not active objects as well. Newly created DDIC objects will be shown in the repo view with an "inactive" icon instead of appearing as non-existing. 

Before: Error msg

image

After: Object is shown with inactive status

image

Todo:

- [x] Remove error message from `zcl_abapgit_object_doma/dtel/tabl/...`

### CODE REVIEW

**Review by larshp — CHANGES_REQUESTED** [review]
see comment above

File: src/objects/zcl_abapgit_objects_super.clas.abap [hunk 1]
```diff
@@ -318,11 +329,16 @@ CLASS zcl_abapgit_objects_super IMPLEMENTATION.
...
+    " ' ' object not in DDIC
+    " 'N' object is in status "new"
+    " 'A' object is in status "active"
+    " 'M' object is in status "revised"
     CALL FUNCTION 'DDIF_STATE_GET'
       EXPORTING
         type          = lv_type
         name          = lv_name
-        state         = 'A'
+        state         = 'M'
```
[the remark this answers is not in the document]
Comment (mbtools) [reply src/objects/zcl_abapgit_objects_super.clas.abap:341]: WIth 'A' you only get 'A' or '' as `gotstate` (active or not). With 'M' you also get the other possible states. The function is actually documented by SAP ;-)

[the remark this answers is not in the document]
Comment (larshp) [reply src/objects/zcl_abapgit_objects_super.clas.abap:341]: suggest adding a comment in the code? I dont think its obvious that M is the correct value here

[the remark this answers is not in the document]
Comment (mbtools) [reply src/objects/zcl_abapgit_objects_super.clas.abap:340]: updated comment

**Review by larshp — APPROVED** [review]

**Pull request merged**